### PR TITLE
bug(hook): useDebouncedCellData fix

### PR DESCRIPTION
**Pull request by avoskresensky**
Fixed useDebouncedCellData as discussed in #948.

Added 2 unit tests to verify the fixes of two outlined problems + another one ("outside changes") to make sure the original corner case logic works as before.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Closing issues

#948

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -10396,6 +10403,15 @@ react-transition-group@^4.4.0:
...
 
+react@^16.8.0:
+  version "16.14.0"
+  resolved "https://registry.yarnpkg.com/react/-/react-16.14.0.tgz#94d776ddd0aaa37da3eda8fc5b6b18a4c9a3114d"
+  integrity sha512-0X2CImDkJGApiAlcf0ODKIneSwBPhqJawOa5wCtKbu7ZECrmS26NvtSILynQ66cgkT/RJ4LidJOc3bUESwmU8g==
+  dependencies:
+    loose-envify "^1.1.0"
+    object-assign "^4.1.1"
+    prop-types "^15.6.2"
+
```
Comment (macrozone) [inline yarn.lock:10414]: this is a dep of jest-react-hooks-shallow, maybe that is outdated?

it probably has no side effect though

**Comment by macrozone** [comment]
@avoskresensky thank you so much for taking the time to tackle this and add a proper test!

i'll check locally and merge into beta. Goal is to release beta soon (next week or so)

**Comment by macrozone** [comment]
there is one problem that  undo might be broken now:

https://user-images.githubusercontent.com/1972353/114039642-373fd500-9883-11eb-834d-7be8335fea00.mov


notice how the undo button goes back to be "at the end" (so no redo is possible).

reason is that after undo, it seems to send out another onChange event that will erase the future

File: yarn.lock (hunk 1 above)
Comment (avoskresensky) [reply yarn.lock:10414]: Those three were already a part of the node_modules. Their versions remained intact.

**Comment by avoskresensky** [comment]
@macrozone , that was a problem inside the slate plugin: onChangeDebounced callback was missing a dependency (`data.slate`) and now that we made `props.onChange` referentially stable, it stopped updating the closure and notUndoable flag was set incorrectly.

I updated the dependency which fixed the immediate problem, but the slate plugin obviously needs some more work:

- The debouncing logic can probably go now that we have it "upstairs"
- There is another example of the dreaded "stale state" anti-pattern:

```
const [value, setValue] = useState<Node[]>(data?.slate);
const valueRef = useRef(value);
valueRef.current = value;
```

**Comment by macrozone** [comment]
works good as far as i can tell! thank you very much @avoskresensky
